### PR TITLE
Find posts by a given date

### PR DESCRIPTION
**Pull request by elvismdev**
We are not sure if here would be the right place/file to add this feature but @lele140686 and me came into this solution for a project where we created a simple API to check wordpress posts and we needed to get the posts for a given date. We would like to submit it for @ekino to review and merge it if this is fine.

### CODE REVIEW

File: Manager/PostManager.php [hunk 1]
```diff
@@ -114,4 +114,19 @@ public function getThumbnailPath(Post $post)
 
         return $post->getGuid();
     }
+    
+    /**
+     * @param \DateTime $date
+     * @return array
+     */
+    public function findByDate(\DateTime $date)
```
Comment (rande) [inline Manager/PostManager.php]: @eko will review this PR, but I think we should accept ``$date = null`` and set it to current date if the value is null.

File: Manager/PostManager.php [hunk 2]
```diff
@@ -114,4 +114,19 @@ public function getThumbnailPath(Post $post)
...
+    {
+        $qb = $this->repository->createQueryBuilder('c');
```
Comment (eko) [inline Manager/PostManager.php]: Can you please just call here a repository method and move this query logic into the repository class?

This way, `PostManager` should have the following code:

```php
    /**
     * Returns posts for current date or a specified date.
     *
     * @param \DateTime $date
     *
     * @return array
     */
    public function findByDate(\DateTime $date = null)
    {
        $date = $date ?: new \DateTime();

        return $this->repository->findByDate($date);
    }
```

and move this query logic into `PostRepository`:

```php    
    /**
     * Returns posts for current date or a specified date.
     *
     * @param \DateTime $date
     *
     * @return array
     */
    public function findByDate(\DateTime $date)
    {
        $qb = $this->createQueryBuilder('c')
            ->select('c')
            ->where('c.date = :date')
            ->addOrderBy('c.id')
            ->setParameter('date', '%' . $date->format('Y-m-d') . '%');

        return $qb->getQuery()->getResult();
    }
```

**Comment by eko** [comment]
Hi @elvismdev,

Thank you for this PR.

Can you please correct the comment I've added and also provide a little unit test for `PostManager` class in order to ensure that the repository is called with the correct date?

Thank you

File: Manager/PostManager.php (hunk 2 above)
Comment (xavismeh) [inline Manager/PostManager.php]: `->setParameter('date', $date->format('Y-m-d') . '%');`should be a bit more efficient :)
Also, the sort should be made by date first, then by ID shouldn't it ?

Comment (eko) [inline Manager/PostManager.php]: Yes, it should be better and I agree for the date sort too. I will update my comment.

**Comment by eko** [comment]
Hi @elvismdev, do you think you will have some time to update this PR?

Thank you

**Comment by elvismdev** [comment]
Hi @eko sorry the delay to get back to review this. I will send an update tomorrow with the recommendations mentioned in the comments. Thanks!

File: Repository/PostRepository.php [hunk 3]
```diff
@@ -21,4 +21,19 @@
  */
 class PostRepository extends EntityRepository
 {
+    /**
+     * @param \DateTime|null $date
+     * @return array
```
Comment (eko) [inline Repository/PostRepository.php]: Can you add a white line between `@return` statement please?

```php
    /**
     * Returns posts for a specific date ordered by descendent date and identifier.
     *
     * @param \DateTime|null $date
     *
     * @return array
     *
     * @author Elvis Morales <elvismdev@gmail.com> Leroy Ley <lele140686@gmail.com> Vincent Composieux <composieux@ekino.com>
     */
```

Thank you

Comment (lele140686) [inline Repository/PostRepository.php]: You also removed my name :(

Comment (eko) [inline Repository/PostRepository.php]: @lele140686, Sorry, that was not voluntary. I've updated my comment.

Comment (elvismdev) [inline Repository/PostRepository.php]: Hi @eko I just added the white line to space between @return. Thank you.

**Comment by eko** [comment]
Thank you @elvismdev

**Comment by elvismdev** [comment]
Awesome @eko! Am glad you like it. I found this project very useful for many things I have on mind for WordPress sites here on my company so hopefully I will be sending more collaborations into the project

**Pull request merged**